packages/headless/src/controllers/insight/category-facet/headless-insight-category-facet.ts

```diff
@@ -0,0 +1,201 @@
+  logFacetShowLess,
+  logFacetShowMore,
+  logFacetUpdateSort,
+} from '../../../features/facets/facet-set/facet-set-insight-analytics-actions';
+import {
+  executeSearch,
+  fetchFacetValues,
+} from '../../../features/insight-search/insight-search-actions';
+import {
+  categoryFacetSearchSet,
+  categoryFacetSet,
+  configuration,
+  search,
+} from '../../../app/reducers';
+import {
+  CategoryFacetSearchSection,
+  CategoryFacetSection,
+  ConfigurationSection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+
+export interface InsightCategoryFacetOptions extends CategoryFacetOptions {}
+
+export interface InsightCategoryFacetProps {
+  /**
+   * The options for  `InsightCategoryFacet` controller
+   */
+  options: InsightCategoryFacetOptions;
+}
+
+export interface InsightCategoryFacet extends Controller {
+  /**
+   * Toggles the specified facet value.
+   *
+   * @param selection - The facet value to toggle.
+   */
+  toggleSelect(selection: CategoryFacetValue): void;
+
+  /**
+   * Deselects all facet values.
+   * */
+  deselectAll(): void;
+
+  /**
+   * Sorts the facet values according to the specified criterion.
+   *
+   * @param criterion - The criterion by which to sort values.
+   */
+  sortBy(criterion: CategoryFacetSortCriterion): void;
+
+  /**
+   * Checks whether the facet values are sorted according to the specified criterion.
+   *
+   * @param criterion - The criterion to compare.
+   * @returns Whether the facet values are sorted according to the specified criterion.
+   */
+  isSortedBy(criterion: CategoryFacetSortCriterion): boolean;
+
+  /**
+   * Increases the number of values displayed in the facet to the next multiple of the originally configured value.
+   */
+  showMoreValues(): void;
+
+  /**
+   * Sets the number of values displayed in the facet to the originally configured value.
+   * */
+  showLessValues(): void;
+
+  /**
+   * Enables the facet. I.e., undoes the effects of `disable`.
+   */
+  enable(): void;
+
+  /**
+   * Disables the facet. I.e., prevents it from filtering results.
+   */
+  disable(): void;
+
+  /**
+   * The state of the `Facet` controller.
+   * */
+  state: InsightCategoryFacetState;
+}
+
+export interface InsightCategoryFacetState extends CoreCategoryFacetState {}
+
+/**
+ * Creates a `InsightCategoryFacet` controller instance.
+ *
+ * @param engine - The insight engine.
+ * @param props - The configurable `InsightCategoryFacet` properties.
+ * @returns A `InsightCategoryFacet` controller instance.
+ * */
+export function buildInsightCategoryFacet(
+  engine: InsightEngine,
+  props: InsightCategoryFacetProps
+): InsightCategoryFacet {
+  const coreController = buildCoreCategoryFacet(engine, props);
+  const {dispatch} = engine;
+  const getFacetId = () => coreController.state.facetId;
+
+  if (!loadInsightCategoryFacetReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  return {
+    ...coreController,
```

Comment:
We probably need to return a facetSearch as well or the quantic component might explode. Would be good to validate and see with @lbergeron about using the same noop controller he added.

Reply:
done